projects/storefrontapp-e2e-cypress/cypress/integration/regression/asm/asm.e2e-spec.ts

```diff
@@ -48,6 +51,62 @@ context('Assisted Service Module', () => {
       cy.log('--> sign out and close ASM UI');
       asm.agentSignOut();
     });
+
+    it('agent should be able to bind anonymous cart to customer', () => {
+      const customer = getSampleUser();
+
+      let anonymousCartCode: string;
+      checkout.visitHomePage();
+      cy.get('cx-asm-main-ui').should('not.exist');
+
+      checkout.registerUser(false, customer);
+
+      cy.log('--> Add to cart as an anonymous user');
+      cart.addProductAsAnonymous();
+
+      cy.log('--> Retrieve cart id');
+      cart.goToCart();
+      cy.get('cx-cart-details')
+        .get('h2.cx-total')
+        .then(($cartId) => {
+          // localStorage contains anonymous cart uid, read code from UI
+          const text = $cartId.text();
+          anonymousCartCode = text.replace('Cart #', '').trim();
```

Comment:
Cart id could perhaps also be retrieved from local storage here as well.
Could doing so also avoid to go to the cart page?

Reply:
The comment on line 72 was intended to explain why this is reading the code from the UI.  The local storage contains the cart uid for an anonymous cart, but an authenticated-user-bound cart the local storage contains the cart code.  We cannot compare them directly.